### PR TITLE
Changed to TypeScript

### PR DESCRIPTION
**Pull request by josephriosIO**
**Context**
We want to publish v1.0 of Snipsync.
These changes are aimed at getting us closer to that.

**Per Joseph**
A few changes I made to snipsync were:

-  convert it from vanilla JavaScript to TypeScript
- Updated the `README.md` 
- Installed new npm packages 
- Installed ESLint

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (flossypurse) [inline tsconfig.json:1]: [Sand] - I don't think we are using this config anymore right? If so, lets just remove it.

File: src/Sync.ts [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+    rimrafAsync(path);
+    cleanupProgress.update(1);
+    cleanupProgress.stop();
+  }
+}
+
+function determineExtension(path) {
+    const parts = path.split(".");
+    return parts[parts.length - 1];
+  }
```
Comment (flossypurse) [inline src/Sync.ts:344]: [Sand] - Lets format these so there is no indentation for these functions.

**Review by flossypurse — APPROVED** [review]
Awesome Joseph. 
Just a couple pieces of sand - otherwise - lgtm.

File: tsconfig.json (hunk 1 above)
Comment (josephriosIO) [reply tsconfig.json:1]: The tsconfig is for configuring how typescript will compile it to javascript on build I believe I could be wrong though

File: src/Sync.ts (hunk 2 above)
Comment (josephriosIO) [reply src/Sync.ts:344]: Changes made!

File: tsconfig.json (hunk 1 above)
Comment (flossypurse) [reply tsconfig.json]: Cool - then you are good to land this!

**Pull request merged**